### DISCUSSION TITLE
Remote Access Firefox SOCKS proxy

### DISCUSSION THREAD

**Discussion by aprilnovak** [idaholab/moose#16445] [Q&A Getting Started] [answered]
Hi all,

I'm trying to get my remote access set up on my new computer (macOS 10.15.7), but am running into some difficulties actually accessing `hpcgitlab.hpc.inl.gov`. I can successfully ssh to `hpclogin.inl.gov`, but the remote access instructions on the MOOSE website don't work for me - I cannot get to `hpclogin.hpc.inl.gov` through Firefox 83.0. I also tried Safari 14.0 and had no luck.

The strange thing is that I have this proxy set up just fine on another computer (macOS 10.14.6), and I'm just straight copying what I did in Firefox's network settings to my new computer. I tried adding no proxy for google.com, and I can get through to it just fine - so it seems that something about `hpclogin.hpc.inl.gov` doesn't work for me.

This may be an issue with Firefox or Mac, but I just wanted to ask the question here in case any other Mac users have this problem? Am I missing a factor of `hpc` somewhere (I vaguely remember something in the past having to do with different website URLs for external vs. internal users)? Is there some permissions I need to change in macOS 10.15.7 that I didn't need before?

Thanks,

-April

**Comment by loganharbour** [comment, marked as answer]
Crazy timing! I'm actually changing some of this right now. Take a look at https://github.com/idaholab/moose/pull/16446.

If you have the current SSH config listed within the instructions, and have a blanket proxy (all traffic to the SOCKS proxy)... it should work. Can you give the new setup a try?

Add the file here https://github.com/idaholab/moose/blob/4e88a478314032399feec8592e6837f7a6009f93/scripts/hpc_proxy.pac anywhere on your filesystem and add it as the "automatic proxy configuration url" in Firefox (`file://some/path/to/hpc_proxy.pac`).

**Reply by loganharbour** [reply]
Oh, I think I missed what you read some. You can't get to _any_ resources within a browser. On first read, I took that as you could get to some resources but not all.

This leads me to believe that it's an issue with the SOCKS proxy as a whole, not how you're forwarding to it.

**Reply by aprilnovak** [reply]
It is crazy timing!

If I did everything correctly, then unfortunately the new approach also doesn't work for me. The only apparent difference is that now Firefox hangs indefinitely on hpcgitlab.hpc.inl.gov, instead of just giving up right away :/

**Reply by loganharbour** [reply]
> The only apparent difference is that now Firefox hangs indefinitely on hpcgitlab.hpc.inl.gov, instead of just giving up right away :/

Yeah, that's because with the new config if the proxy fails it'll try a direct connection (for people on INL VPN). Which means... something is definitely borked with the SOCKS proxy.

Are you connecting to hpclogin with
```
ssh hpclogin
```
?

If you're connecting using the FQDN (hpclogin.inl.gov), it won't setup the proxy.

**Reply by aprilnovak** [reply]
That fixed it! I had copied over my bash script with a few aliases to make that ssh process shorter, so I didn't notice that what I was doing before (`ssh novaapri@hpclogin.inl.gov`) didn't gel with the current MOOSE access instructions. I guess I didn't expect that part to change.

Thanks, Logan!
